File: src-electron/cli/detector/configs.ts
/**
 * Common config files detector (Docker Compose, Justfile, Taskfile)
 */

import * as fs from 'fs';
import * as path from 'path';
import type { ProjectTemplate } from '../types';

export function detectCommonConfigs(directory: string): ProjectTemplate[] {
  const projects: ProjectTemplate[] = [];

  // Docker Compose
  const dockerComposeFiles = [
    'docker-compose.yml', 'docker-compose.yaml',
    'compose.yml', 'compose.yaml'
  ];

  for (const file of dockerComposeFiles) {
    if (fs.existsSync(path.join(directory, file))) {
      projects.push({
        name: 'docker: compose up',
        command: `docker compose -f ${file} up`,
        description: 'Start Docker Compose services',
        priority: 20
      });
      break;
    }
  }

  // Justfile
  if (fs.existsSync(path.join(directory, 'justfile')) ||
      fs.existsSync(path.join(directory, 'Justfile'))) {
    projects.push({
      name: 'just: list',
      command: 'just --list',
      description: 'List just recipes',
      priority: 25
    });
  }

  // Taskfile
  if (fs.existsSync(path.join(directory, 'Taskfile.yml')) ||
      fs.existsSync(path.join(directory, 'Taskfile.yaml'))) {
    projects.push({
      name: 'task: list',
      command: 'task --list',
      description: 'List task targets',
      priority: 25
    });
  }

  return projects;
}

File: src-tauri/src/cli/detector/configs.rs
use crate::cli::detector::ProjectTemplate;
use std::path::Path;

/// Detect common config files
pub fn detect_common_configs(directory: &Path) -> Vec<ProjectTemplate> {
    let mut projects = Vec::new();

    // Docker Compose
    let docker_compose_files = [
        "docker-compose.yml",
        "docker-compose.yaml",
        "compose.yml",
        "compose.yaml",
    ];

    for file in &docker_compose_files {
        if directory.join(file).exists() {
            projects.push(ProjectTemplate {
                name: "docker: compose up".to_string(),
                command: format!("docker compose -f {} up", file),
                description: "Start Docker Compose services".to_string(),
            });
            break;
        }
    }

    // Justfile
    if directory.join("justfile").exists() || directory.join("Justfile").exists() {
        projects.push(ProjectTemplate {
            name: "just: list".to_string(),
            command: "just --list".to_string(),
            description: "List just recipes".to_string(),
        });
    }

    // Taskfile
    if directory.join("Taskfile.yml").exists() || directory.join("Taskfile.yaml").exists() {
        projects.push(ProjectTemplate {
            name: "task: list".to_string(),
            command: "task --list".to_string(),
            description: "List task targets".to_string(),
        });
    }

    projects
}
